### PR TITLE
Matter Pump: add support for pump device type

### PR DESCRIPTION
**Pull request by ctowns**
These matter pump changes were taken from this Matter 1.2 PR so we could address some remaining open items: https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/pull/1422

### CODE REVIEW

File: drivers/SmartThings/matter-pump/profiles/pump-level.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  - id: pumpOperationMode
+    version: 1
+  - id: pumpControlMode
+    version: 1
+  - id: firmwareUpdate
+    version: 1
+  - id: refresh
+    version: 1
+  categories:
+  - name: Irrigation
```
Comment (ctowns) [inline drivers/SmartThings/matter-pump/profiles/pump-level.yml:18]: Shouldn't we use the Pump category @HunsupJung ?

Comment copied from here: https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/pull/1422#discussion_r1632061585

File: drivers/SmartThings/matter-pump/profiles/pump-level.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+metadata:
```
Comment (ctowns) [inline drivers/SmartThings/matter-pump/profiles/pump-level.yml:19]: @HunsupJung why is pump using a custom presentation?

Comment copied from here: https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/pull/1422#discussion_r1632061902

File: drivers/SmartThings/matter-pump/profiles/pump-only.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    version: 1
+  - id: firmwareUpdate
+    version: 1
+  - id: refresh
+    version: 1
+  categories:
+  - name: Irrigation
+metadata:
+  mnmn: SmartThingsEdge
+  vid: generic-pump-level
```
Comment (tpmanley) [inline drivers/SmartThings/matter-pump/profiles/pump-only.yml:19]: If a custom vid is needed, is this the core one? This profile doesn't have level

File: drivers/SmartThings/matter-pump/profiles/pump-level.yml (hunk 1 above)
Comment (HunsupJung) [reply drivers/SmartThings/matter-pump/profiles/pump-level.yml:18]: HJ commented what the process was on following ticket.
https://smartthings.atlassian.net/browse/CHAD-13354?focusedCommentId=1456242

File: drivers/SmartThings/matter-pump/profiles/pump-level.yml (hunk 2 above)
Comment (HunsupJung) [reply drivers/SmartThings/matter-pump/profiles/pump-level.yml:19]: It's related naming of switchLevel capability. switchLevel Capability is displayed as `dimmer`. so we needed to modify it to `level` through device configuration.

File: drivers/SmartThings/matter-pump/profiles/pump-only.yml (hunk 3 above)
Comment (HunsupJung) [reply drivers/SmartThings/matter-pump/profiles/pump-only.yml:19]: The reason why the same device configuration is applied to all profiles is that the presentation id of the card is changed when a profile which device configuration is applied is changed from a profile which device configuration is not applied in the driver. and There is no way to check if the presentation id has been changed in the app, so we applied device configuration to all profiles. this is the way proposed by the presentation team, so please let me know if there is a better way.

**Review by HunsupJung — APPROVED** [review]

File: drivers/SmartThings/matter-pump/profiles/pump-level.yml (hunk 2 above)
Comment (tpmanley) [reply drivers/SmartThings/matter-pump/profiles/pump-level.yml:19]: I see. Seems like that's the best option for now but let's talk internally about longer term options so this isn't necessary

File: drivers/SmartThings/matter-pump/profiles/pump-only.yml (hunk 3 above)
Comment (tpmanley) [reply drivers/SmartThings/matter-pump/profiles/pump-only.yml:19]: I don't think I was aware of this issue. Let's talk about this internally to see if there are other options

File: drivers/SmartThings/matter-pump/profiles/pump-level.yml (hunk 1 above)
Comment (tpmanley) [reply drivers/SmartThings/matter-pump/profiles/pump-level.yml:18]: I responded on the ticket and we can discuss further there

File: drivers/SmartThings/matter-pump/profiles/pump-level.yml (hunk 2 above)
Comment (HunsupJung) [reply drivers/SmartThings/matter-pump/profiles/pump-level.yml:19]: @tpmanley 
Yes, please tell me if there is any part that needs to be proceeded.

File: drivers/SmartThings/matter-pump/profiles/pump-level.yml (hunk 1 above)
Comment (ctowns) [reply drivers/SmartThings/matter-pump/profiles/pump-level.yml]: I have updated the profiles to use the `Pump` category based on internal discussion here: https://smartthings.atlassian.net/browse/CHAD-13354?focusedCommentId=1456242

File: drivers/SmartThings/matter-pump/profiles/pump-level.yml (hunk 2 above)
Comment (ctowns) [reply drivers/SmartThings/matter-pump/profiles/pump-level.yml:19]: Let's plan to wrap up discussion on this topic this week so we can get this driver to beta next week.

Comment (ctowns) [reply drivers/SmartThings/matter-pump/profiles/pump-level.yml:19]: I think we will need to move forward with the metadata for now to modify the presentation to say `level`

File: drivers/SmartThings/matter-pump/profiles/pump-only.yml (hunk 3 above)
Comment (ctowns) [reply drivers/SmartThings/matter-pump/profiles/pump-only.yml:19]: @HunsupJung does this mean that both profiles will use the `generic-pump-level` metadata, but the level will still not be displayed for the `pump-only` profile?

Comment (HunsupJung) [reply drivers/SmartThings/matter-pump/profiles/pump-only.yml:19]: Yes, that is correct.

Comment (ctowns) [reply drivers/SmartThings/matter-pump/profiles/pump-only.yml:19]: If we move forward with the custom metadata, will be we able to set things like `capabilities.switchLevel.levelRange()` in the driver, or will the metadata override that? This is similar to the question I asked here: https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/pull/1458#discussion_r1653382008

Comment (ctowns) [reply drivers/SmartThings/matter-pump/profiles/pump-only.yml:19]: I have tested this and the answer is yes, we will still be able to use things like the `capabilities.switchLevel.levelRange()` as needed

**Pull request merged**